### PR TITLE
DDCYLS-1150

### PR DESCRIPTION
**Pull request by dips7189**
MongoPersistence - Added function in registrations to find by DST Registration Number
Persistence - Added function in registrations to find by DST Registration Number
FutureVolatilePersistence - Added function in registrations to find by DST Registration Number
VolatilePersistence - Added function in registrations to find by DST Registration Number
StartUpModule - eager binding to start DST Registration Number checker on start up
DstRegChecker - checker to check the DST Registration exists

### CODE REVIEW

**Review by wp94 — APPROVED** [review]

File: app/uk/gov/hmrc/digitalservicestax/services/DstRegChecker.scala [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    val dstRegNumber: String @@ data.DSTRegNumber.Tag = optDstRegNumber.head
+
+    /**
+     * Registrations
+     */
+    val effOptReg: Future[Option[Registration]] = db.registrations.findByDstReg(dstRegNumber)
+
+    effOptReg.map { optReg =>
+      if (optReg.isEmpty) {
+        logger.error("\n<<>><<>><<>><<>><<>><<>><<>>ERROR NO REGISTRATION COULD BE FOUND FOR THE INTERNAL ID<<>><<>><<>><<>><<>><<>><<>>\n")
```
Comment (ashwinirudrappa) [inline app/uk/gov/hmrc/digitalservicestax/services/DstRegChecker.scala:54]: @dips7189 only logger.warn is used in prod

File: test/unit/uk/gov/hmrc/digitalservicestax/services/VolatilePersistence.scala [hunk 2]
```diff
@@ -50,6 +50,7 @@ trait VolatilePersistence extends Persistence[Id] {
     def update(user: InternalId, reg: Registration): Unit = 
       _data = _data + (user -> ((reg, LocalDateTime.now)))
 
+    override def findByDstReg(DSTRegNumber: DSTRegNumber): Id[Option[Registration]] = ???
```
Comment (ashwinirudrappa) [inline test/unit/uk/gov/hmrc/digitalservicestax/services/VolatilePersistence.scala:53]: @dips7189 Do we not need this code to be implemented

**Review by ashwinirudrappa — APPROVED** [review]

File: app/uk/gov/hmrc/digitalservicestax/services/DstRegChecker.scala (hunk 1 above)
Comment (dips7189) [reply app/uk/gov/hmrc/digitalservicestax/services/DstRegChecker.scala:54]: Thank you, I 've changed this now :)

File: test/unit/uk/gov/hmrc/digitalservicestax/services/VolatilePersistence.scala (hunk 2 above)
Comment (dips7189) [reply test/unit/uk/gov/hmrc/digitalservicestax/services/VolatilePersistence.scala:53]: To be honest, thet tests & everything are running fine with out having done anything with this.
looking at what it does, it has a map that maintains registrations like a cache would.
It's a var that has had `@volatile` put on there.
From what I understand the user that is having issues should already be in the database, hence I left the implementation for this apparent cache blank.

**Pull request merged**